modules/misc/src/postprocessors/ThermocoupleSensorPostprocessor.C

```diff
@@ -0,0 +1,303 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ThermocoupleSensorPostprocessor.h"
+
+registerMooseObject("MooseApp", ThermocoupleSensorPostprocessor);
+
+InputParameters
+ThermocoupleSensorPostprocessor::validParams()
+{
+  InputParameters params = GeneralSensorPostprocessor::validParams();
+  params.addParam<string>("thermocouple_type", "B", "The type of thermocouple (B, E, J or K)");
+  params.addParam<string>("method", "numerical", "The equation to be used (lumped or numerical).");
+  params.addClassDescription("This is a ThermocoupleSensorPostprocessor for various classes of "
+                             "thermocouples, described by the 'thermocouple_type' parameter");
+  params.addParam<Real>("proportional_weight", 0, "The weight assigned to the proportional term");
+  params.addParam<Real>("integral_weight", 1, "The weight assigned to the integral term");
+  return params;
+}
+
+ThermocoupleSensorPostprocessor::ThermocoupleSensorPostprocessor(const InputParameters & parameters)
+  : GeneralSensorPostprocessor(parameters),
+  _thermocouple_type(getParam<string>("thermocouple_type")),
+  _method(getParam<string>("method"))
+{
+}
+
+void ThermocoupleSensorPostprocessor::initialize() 
+{ 
+  Real drift_value;
+  Real efficiency_value;
+  Real signalToNoise_value;
+  Real noise_value;
+  Real noise_std_dev;
+  Real uncertainty_value;
+  Real uncertainty_std_dev;
+  Real delay_value;
+
+  // setting values for thermocouple types B, J, E, K
+  // current values are placeholders
+  if (_thermocouple_type == "B")
+  {
+    // B type, temperature in degree celcius
+    if (_input_signal >= 0 && _input_signal < 630.615)
+    {
+      drift_value = 1;
```

Comment:
```suggestion
      drift_value = 1.0;
```

you are assigning an integer literal to a double precision float variable. The compiler converts that for you, but IMO it's better style to use a float literal (e.g. add the decimal point)